openassessment/xblock/submission_mixin.py

```diff
@@ -410,34 +410,59 @@ def upload_url(self, data, suffix=''):  # pylint: disable=unused-argument
             A URL to be used to upload content associated with this submission.
 
         """
+        # Validate that there are no data issues and file type is allowed
+        try:
+            if not self.is_supported_upload_type(data):
+                return {'success': False, 'msg': self._(
+                    "File upload failed: unsupported file type."
+                    "Only the supported file types can be uploaded."
+                    "If you have questions, please reach out to the course team."
+                )}
+        except KeyError:
+            return {'success': False, 'msg': self._("There was an error uploading your file.")}
+
+        # Attempt to upload
+        file_num = int(data.get('filenum', 0))
```

Comment:
nit: we could pass these into is_supported_upload_type rather than repeating those two lines

Reply:
Which would we pass in?

Reply:
Good call. Refactored.